### DISCUSSION TITLE
New version of Rallly incoming

### DISCUSSION THREAD

**Discussion by lukevella** [lukevella/rallly#702] [Announcements]
Hi everyone,

I'm going to be releasing Rallly v3 soon and wanted to share this with you first as I'd like to gather feedback from the community. The new version is currently running in parallel to the old version.

[Try it out &rarr;](https://app.rallly.co)

## What's new?

I made it a priority not to change too much of the existing workflow since this would add a lot of overhead and risks when transitioning between versions so the app should feel familiar. That said, some key changes:

### Separate App from Landing Page 

Previously, the landing page and the app where in a single nextjs. This was done purely out of convenience very early on in the project but quickly became a regret as it made it harder to separate concerns. This means the new version will be hosted on [app.rallly.co](https://app.rallly.co]) rather than [rallly.co](https://rallly.co) and for self-hosters, it means that no configuration is required to disable the landing page.

![CleanShot 2023-06-17 at 11 38 20](https://github.com/lukevella/rallly/assets/676849/5732b717-0fe6-4656-bab6-85e1a20245b6)

### Updated UI and Navigation

I've added some proper navigation UI so that we can add new features. This was a major limitation in the old design that wasn't really intended to grow to accommodate more features.

![CleanShot 2023-06-17 at 11 38 39](https://github.com/lukevella/rallly/assets/676849/24c940f2-cb32-43fd-b66f-c29f7d2762d7)

### Rallly Pro

We've come a long way with the help of sponsors and donations but without any revenue it won't be possible for me to continue working on Rallly full-time as I have been. So I will be introducing a paid tier called Rallly Pro. I plan on delivering loads of new features that will be availably exclusively to Pro users. I'll make a blog post soon outlining the vision/roadmap but the first feature that will be available to Pro users is the ability to finalize polls and send calendar invites.

![CleanShot 2023-06-17 at 11 39 00](https://github.com/lukevella/rallly/assets/676849/4d221d62-5cb6-4828-a07e-d691a3b3da11)

### Poll Status

Rather than locking and unlocking a poll, in the new version you will be able to pause and resume a poll. This terminology fits in better with the ability to finalize a poll and the behaviour maps well between version. i.e. a locked poll will be paused in the new version.

![CleanShot 2023-06-17 at 11 37 13](https://github.com/lukevella/rallly/assets/676849/ddce6206-e3d5-432e-92ca-5fc220a6d5f8)

## Self-Hosters

I plan to release a docker image for self-hosters after the official release of the new version. My plan is to make all Pro features available for free to self-hosted instances but perhaps limit these instances to single-users. For self-hosted instances with multiple users (presumably used by companies), it would make sense to turn this into a revenue stream so I'd like to offer multi-user licenses that will unlock this functionality. That said, I currently don't have any insight on whether there is enough demand to justify the work needed to do this. If you or your company uses Rallly, I'd love to hear from you.

## Conclusion

That's it for now. Thanks for following along and I really hope you like the new version. Please share your thoughts here and let me know if you have any questions.

**Comment by davidlohner** [comment]
I know that this announcement is not the newest, as v3.0 just shipped (thanks for that and great work!)

I can fully understand your thoughts on the Pro features, pricing and availability on self-hosted instances. I'd like to suggest considering  multi-user instances should be allowed for non-profit organizations.

**Reply by lukevella** [reply]
Yeh my plan changed since making this announcement. The self-hosted version is completely free for everyone. Things may change for new versions but I'm in favor of keeping it free for non-profits 👍

**Reply by davidlohner** [reply]
Thats great to hear, thank you! 
How do I unlock the pro features on my self-hosted Rallly instance? (which I'm running via cloudron.io)

**Reply by lukevella** [reply]
I think they might have to update to the latest image then. You're likely still on an older version.

**Reply by jdaviescoates** [reply]
I'm confused, the [Cloudron Rallly appears to be on 3.0.0](https://forum.cloudron.io/post/73584) but 1) the Pro features aren't available and 2) this issue https://github.com/lukevella/rallly/commit/8e7b5a3d93d9c0779605532b3e9ee1ceadef2c9e seems to still exist

**Reply by lukevella** [reply]
It sounds like you're running the [3.0.0-rc1](https://hub.docker.com/layers/lukevella/rallly/3.0.0-rc1/images/sha256-9921061110c5f1daecfbe94e882466ecca44f3cb8d041bdf2b82e74d3de3224b?context=explore) image from docker hub.

I'm not sure how cloudron manages these updates to be honest. I'll try releasing a patch and see if it will automatically update.

**Reply by jdaviescoates** [reply]
That's great, thanks! :pray:  I don't think it's automatic but the Cloudron devs @gramakri and @nebulade are normally pretty quick...

**Reply by jdaviescoates** [reply]
Ah, seems like @nebulade is already on it anyway https://forum.cloudron.io/post/73726

**Reply by lukevella** [reply]
Thanks for pointing me to that post. @jdaviescoates, could you clarify what you meant when you say the Pro features are not available? Do you mean that they are locked or you do not see them?

**Reply by jdaviescoates** [reply]
Sure, we see them, but they are locked:

![image](https://github.com/lukevella/rallly/assets/2113743/28a2f52a-87a8-4b4e-b87c-88cf29d217d2)

![image](https://github.com/lukevella/rallly/assets/2113743/2ff53686-2ea3-4c45-bfab-921a128c55a3)

But, I think this is likely just because, as @nebulade just said over on the Cloudron forum:

>[Our e2e tests need a bit of fixing due to the new auth situation, so it may take a bit longer than usual for the package update.](https://forum.cloudron.io/post/73729)

and 

> [we have not set the selfhosted environment variable in the past. This comes with the next package update.](https://forum.cloudron.io/post/73730)

:slightly_smiling_face:

**Reply by lukevella** [reply]
I see… that's not expected when running the published docker images. My concern is that if the image is being built from source it may be relying on APIs that were not intended for public consumption. There's no guarantees that it will work if it is being built from source.

**Reply by nebulade** [reply]
Hi, just to update here to avoid further work on your side, we have not set the `NEXT_PUBLIC_SELF_HOSTED` env variable so far, so there might be various side-effects happening which are not worth looking into for the moment.

**Reply by lukevella** [reply]
Hi @nebulade, can you confirm whether or not you are building your own image from source? I'd advise against doing that to avoid these kinds of issues. I would recommend using the published docker hub images. If you pin to a major version you will avoid pulling in breaking changes accidentally.

The main branch is not guaranteed to be working for self-hosters.

**Reply by nebulade** [reply]
Yes we are building our own images, for all apps actually. This has to do with how Cloudron as a platform works. For example all apps run in a read-only filesystem as well as use shared instances of databases and such. We require this to provide solid update/restore functionality, and it is based on our long-time experience with upstream docker images. Further this ensures apps share the same baseimage which is required for our tooling and debugging purposes. While we do use docker, this is only a technical detail from our perspective. We are aware of the extra overhead on our side though.

For reference the calcom package code is at https://git.cloudron.io/cloudron/calcom-app (mostly Dockerfile and start.sh are the relevant resources here) any hints on improvement here are very welcome :)

**Reply by lukevella** [reply]
Thanks for sharing, I'll try and keep this in mind then.

Likewise if you have any suggestions on how this can be made easier for you to work with, let me know.

**Comment by ltguillaume** [comment]
> My plan is to make all Pro features available for free to self-hosted instances but perhaps limit these instances to single-users.

@lukevella I sincerely hope you'll find another way to make the Pro tier interesting: while I use my instance exclusively for me, I often just ask other people to suggest dates this way, even for 1:1 meet-ups. I also tend to set up polls for groups, but if someone else picks up the organization part, I just hand over the poll to that user with a bit of SQL (this could be a nice new feature, tough, granting multiple participants "user" rights).

**Comment by samsapti** [comment]
Hey @lukevella, just wanted to add my 2 cents here.

We at [data.coop](https://data.coop/en/) host Rallly as a public service. We're a democratically run association based in Copenhagen, Denmark, and fully owned and funded by our members. Our core values are, among other things, online privacy, decentralization and FOSS. We own our own data on our own server infrastructure, and offer various services for our members (and some for non-members, fx. Rallly).

While I understand your points and that you would like to earn some revenue for developing Rallly, I would like to point out a couple of things.

Firstly, I would highly suggest keeping it free for non-profit organizations. I know this has already been suggested and that you've stated that you will keep it free, and I appreciate it and appreciate the work you're putting into developing Rallly. With that said, you're of course free to do whatever you want with the instance you host at [rallly.co](https://app.rallly.co).

The other thing I wanted to point out, is this: I'm not sure how well restricting self-hosted instances / putting features behind a paywall plays with the AGPL license that Rallly is released under. According to that, licensees are free to do what they want with the software, including redistributing it. There are examples of paid (A)GPL licensed software (fx. RHEL or Cryptomator), but technically anybody would be able to take the source code, compile and package it and distribute it for free. I'm not saying you can't put features behind a license, but you just need to be aware of this.

With that said, thank you for your effort and time put into Rallly, it really is a great product.
